### PR TITLE
providers/google: Add google_compute_image resource

### PR DESCRIPTION
**Pull request by evandbrown**
This change introduces the google_compute_image resource, which allows
Terraform users to create a bootable VM image from a raw disk tarball
stored in Google Cloud Storage. The google_compute_image resource
may be referenced as a boot image for a google_compute_instance.

### CODE REVIEW

File: builtin/providers/google/resource_compute_image.go [hunk 1]
```diff
@@ -78,8 +84,15 @@ func resourceComputeImageCreate(d *schema.ResourceData, meta interface{}) error
 
 	// Build the image
 	image := &compute.Image{
-		Name:        d.Get("name").(string),
-		Description: d.Get("description").(string),
+		Name: d.Get("name").(string),
+	}
+
+	if v := d.Get("description"); v != nil {
```
Comment (radeksimko) [inline builtin/providers/google/resource_compute_image.go]: More idiomatic way to achieve the same thing may be

```go
if v, ok := d.GetOk("description"); ok {
```

😉

**Comment by stack72** [comment]
Hi @evandbrown 

If you could rebase this, then we can get it tested and merged :)

Thanks

Paul

**Comment by evandbrown** [comment]
@stack72 Done!

Output of relevant acceptance tests:

```
$ make testacc TEST=./builtin/providers/google GOOGLE_PROJECT=redacted GOOGLE_REGION=us-central1 TESTARGS='-run=TestAccComputeImage...'
==> Checking that code complies with gofmt requirements...
/home/evanbrown/dev/go-workspaces/terraform/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/08/07 21:31:24 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccComputeImage... -timeout 120m
=== RUN   TestAccComputeImage_basic
--- PASS: TestAccComputeImage_basic (75.34s)
PASS
ok      github.com/hashicorp/terraform/builtin/providers/google 75.466s
```

**Comment by stack72** [comment]
@evandbrown looks like the build fails - looks like there are go vet issues

P.

**Comment by stack72** [comment]
@evandbrown ping - we have some go vet errors :)

```
# github.com/hashicorp/terraform/builtin/providers/google
builtin/providers/google/resource_compute_image.go:118: not enough arguments in call to computeOperationWaitGlobal
builtin/providers/google/resource_compute_image.go:169: not enough arguments in call to computeOperationWaitGlobal
```

**Comment by evandbrown** [comment]
@stack72 That's embarrassing! Just rebased a fix that should fix...

**Comment by stack72** [comment]
spot on :) Thanks! running tests now

**Comment by stack72** [comment]
```
% make testacc TEST=./builtin/providers/google TESTARGS='-run=TestAccComputeImage'           1 ↵
==> Checking that code complies with gofmt requirements...
/Users/stacko/Code/go/bin/stringer
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/08/10 17:11:13 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/google -v -run=TestAccComputeImage -timeout 120m
=== RUN   TestAccComputeImage_basic
--- PASS: TestAccComputeImage_basic (95.12s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/google	95.132s
``` 

:)

**Comment by stack72** [comment]
Will merge when travis goes green!

**Pull request merged**